app/src/main/java/com/android/unio/ui/event/EventCreation.kt

```diff
@@ -1,20 +1,379 @@
+        placeholder = { Text(context.getString(R.string.event_creation_placeholder_time_input)) },
+        label = { Text(timeString) })
+  }
+
+  if (isDatePickerVisible) {
+    DatePickerModal(
+        onDateSelected = {
+          selectedDate = it
+          isDatePickerVisible = false
+        },
+        onDismiss = { isDatePickerVisible = false })
   }
+
+  if (isTimePickerVisible) {
+    TimePickerModal(
+        onTimeSelected = {
+          selectedTime = it
+          isTimePickerVisible = false
+        },
+        onDismiss = { isTimePickerVisible = false })
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DatePickerModal(onDateSelected: (Long?) -> Unit, onDismiss: () -> Unit) {
+  val datePickerState = rememberDatePickerState()
+  val context = LocalContext.current
+
+  DatePickerDialog(
+      onDismissRequest = onDismiss,
+      confirmButton = {
+        TextButton(
+            onClick = {
+              onDateSelected(datePickerState.selectedDateMillis)
+              onDismiss()
+            }) {
+              Text(context.getString(R.string.event_creation_dialog_ok))
+            }
+      },
+      dismissButton = {
+        TextButton(onClick = onDismiss) {
+          Text(context.getString(R.string.event_creation_dialog_cancel))
+        }
+      }) {
+        DatePicker(state = datePickerState)
+      }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TimePickerModal(onTimeSelected: (Long?) -> Unit, onDismiss: () -> Unit) {
+  val timePickerState = rememberTimePickerState(is24Hour = true)
+
+  TimePickerDialog(
+      onDismiss = onDismiss,
+      onConfirm = {
+        onTimeSelected(
+            timePickerState.hour * 60 * 60 * 1000L - 3600000 + timePickerState.minute * 60 * 1000L)
+        onDismiss()
+      }) {
+        TimeInput(state = timePickerState)
+      }
+}
+
+/**
+ * A Dialog that is the analog of the DatePickerDialog, but for TimePicker as it currently does not
+ * exist in the Material3 library.
+ */
+@Composable
+fun TimePickerDialog(
+    onDismiss: () -> Unit,
+    onConfirm: () -> Unit,
+    content: @Composable () -> Unit
+) {
+  val context = LocalContext.current
+  AlertDialog(
+      onDismissRequest = onDismiss,
+      dismissButton = {
+        TextButton(onClick = { onDismiss() }) {
+          Text(context.getString(R.string.event_creation_dialog_cancel))
+        }
+      },
+      confirmButton = {
+        TextButton(onClick = { onConfirm() }) {
+          Text(context.getString(R.string.event_creation_dialog_ok))
+        }
+      },
+      text = { content() })
+}
+
+fun convertMillisToDate(millis: Long): String {
+  val formatter = SimpleDateFormat("dd/MM/yy", Locale.getDefault())
+  return formatter.format(Date(millis))
+}
+
+fun convertMillisToTime(millis: Long): String {
+  val formatter = SimpleDateFormat("HH:mm", Locale.getDefault())
+  return formatter.format(Date(millis))
```

Comment:
If I remember correctly @Aurelien9Code had written some similar code for time -> date conversion somewhere. We should consider extracting these functions in an utility file.

Reply:
Don't think we do. It's alright we can do it in a different PR later since it should only concern one other file